cli/src/cmd/create.rs

```diff
@@ -0,0 +1,199 @@
+//! Create command
+
+use crate::{
+    cmd::{build::BuildArgs, Cmd},
+    opts::{EthereumOpts, WalletType},
+};
+use cast::SimpleCast;
+use ethers::{
+    abi::{Contract, Function, FunctionExt},
+    prelude::{
+        artifacts::{BytecodeObject, Source, Sources},
+        ContractFactory, Http, Middleware, MinimalCombinedArtifacts, Project, ProjectCompileOutput,
+        Provider, Signer, SignerMiddleware,
+    },
+    solc::cache::SolFilesCache,
+};
+use eyre::Result;
+
+use crate::opts::forge::ContractInfo;
+use std::{path::PathBuf, sync::Arc};
+use structopt::StructOpt;
+
+#[derive(Debug, Clone, StructOpt)]
+pub struct CreateArgs {
+    #[structopt(long, help = "constructor args calldata arguments.")]
+    constructor_args: Vec<String>,
+
+    #[structopt(flatten)]
+    opts: BuildArgs,
+
+    #[structopt(flatten)]
+    eth: EthereumOpts,
+
+    #[structopt(help = "contract source info `<path>:<contractname>` or `<contractname>`")]
+    contract: ContractInfo,
+
+    #[structopt(long, help = "verify on Etherscan")]
+    verify: bool,
+}
+
+impl Cmd for CreateArgs {
+    type Output = ();
+
+    fn run(self) -> Result<Self::Output> {
+        // Find Project & Compile
+        let project = self.opts.project()?;
+        let compiled = project.compile()?;
+
+        if self.verify && self.contract.path.is_none() {
+            eyre::bail!("verifying requires giving out the source path");
+        }
+
+        if compiled.has_compiler_errors() {
+            // return the diagnostics error back to the user.
+            eyre::bail!(compiled.to_string())
+        }
+
+        // Get ABI and BIN
+        let (abi, bin) = match self.contract.path {
+            Some(ref path) => self.get_artifact_from_path(&project, path.clone())?,
+            None => self.get_artifact_from_name(compiled)?,
+        };
+
+        // Add arguments to constructor
+        let provider = Provider::<Http>::try_from(self.eth.rpc_url.as_str())?;
+        let constructor_with_args = self.build_constructor_with_args(abi.clone())?;
+
+        // Deploy with signer
+        let rt = tokio::runtime::Runtime::new().expect("could not start tokio rt");
+        let chain_id = rt.block_on(provider.get_chainid())?;
+        if let Some(signer) = rt.block_on(self.eth.signer_with(chain_id, provider))? {
+            match signer {
+                WalletType::Ledger(signer) => {
+                    self.deploy(abi, bin, constructor_with_args, signer)?;
+                }
+                WalletType::Local(signer) => {
+                    self.deploy(abi, bin, constructor_with_args, signer)?;
+                }
+                WalletType::Trezor(signer) => {
+                    self.deploy(abi, bin, constructor_with_args, signer)?;
+                }
+            }
+        } else {
+            eyre::bail!("could not find artifact")
+        }
+
+        Ok(())
+    }
+}
+
+impl CreateArgs {
+    /// Find using only ContractName
+    fn get_artifact_from_name(
```

Comment:
@mattsse wdyt of this way of finding the artifacts? Would it not be easier to call `[.find`](https://github.com/gakonst/ethers-rs/blob/4c677933ce4407fb1307f9077aefc31b9e811cdb/ethers-solc/src/lib.rs#L818)?

Reply:
[.find] only gives out the first it finds, and we need to check if there is more than one with the same name